avm/src/vm/opcode.rs

```diff
@@ -2866,7 +2868,7 @@ pub static OPCODES: Lazy<HashMap<i64, ByteOpcode>> = Lazy::new(|| {
       let mut res_hm = HandlerMemory::new(None, 3);
       res_hm.init_fractal(0);
       res_hm.push_fixed(0, res.status().as_u16() as i64);
-      HandlerMemory::transfer(&headers_hm, 0, &mut res_hm, CLOSURE_ARG_MEM_START);
+      HandlerMemory::transfer(&headers_hm, CLOSURE_ARG_MEM_START, &mut res_hm, CLOSURE_ARG_MEM_START);
```

Comment:
What I've done here is add another bit of nesting; before we were accidentally grabbing the first record of the array and returning it as if it was the whole array, eg instead of getting `Array<KeyVal<string, string>>` it was getting `KeyVal<string, string>`.

That also explains why the test of the `length()` was `2` in the test suite, because `KeyVal<string, string>` being treated as an array would show up with 2 elements.